django/test/client.py

```diff
@@ -494,26 +507,37 @@ def post(
             content_type,
             secure=secure,
             headers=headers,
+            query_params=query_params,
             **extra,
         )
 
-    def head(self, path, data=None, secure=False, *, headers=None, **extra):
+    def head(
+        self, path, data=None, secure=False, *, headers=None, query_params=None, **extra
+    ):
         """Construct a HEAD request."""
-        data = {} if data is None else data
+        if query_params and data:
+            raise ValueError("You cannot specify query_params and data together.")
+        query_params = query_params or data
```

Comment:
This should favour `data`, as per the form example in https://github.com/django/django/pull/17447/files/a261b7669935faa3d27fee2a03ab6f45436a8bd3#r1397428841